SocialNetwork/SocialNetwork.Core/Services/UserService.cs

```diff
@@ -1,36 +1,63 @@
 ﻿using SocialNetwork.Core.Interfaces;
 using SocialNetwork.Core.Models;
-
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
 namespace SocialNetwork.Core.Services;
-
 public class UserService : IUserService
 {
-    private readonly IRepository _repository;
-
-    public UserService(IRepository repository)
+    // TODO repository
+    public List<User> Users {get; set; }
+    public UserService()
     {
-        _repository = repository;
+        Users = new List<User>();
     }
-
-    public IEnumerable<User> GetAllUsers(int skip = 0, int take = 20)
+    public List<User> GetUsers() 
     {
-        return _repository.GetAll<User>()
-            .Skip(skip)
-            .Take(take);
+        return Users;
     }
-
-    public async Task<User> AddUser(User user)
+    public User? GetUserById(int id)
     {
-        throw new NotImplementedException();
+        User? user = Users.FirstOrDefault(unit => unit.Id == id);
+        return user;
     }
-
-    public Task<User> UpdateUser(User user)
+    public User? GetUserByName(string name)
     {
-        throw new NotImplementedException();
+        User? user = Users.FirstOrDefault(unit => unit.Nickname == name);
+        return user;
     }
-
-    public Task DeleteUser(int id)
+    public void UpdateUser(User user)
     {
-        throw new NotImplementedException();
+        if (!UserIsValid(user))
```

Comment:
Rename: IsUserValid
Method with return type bool or boolean variables's names should start with 'is' or 'can'